File: computer_alerts-fe/src/features/articles/models/ArticleRequestModel.ts
export interface ArticleRequestModel {
  articleId: string; // Top-level property
  title: string;
  body: string;
  wordCount: number;
  articleStatus: string;
  requestCount: number;
  likeCount: number;
  category: string;
  tagsTag: string;
  timePosted: string;
  photoUrl: string;
  authorIdentifier: string;
  articleDescpition: string;
}

export interface ArticleRequestModelI {
  fileName: string | number | readonly string[] | undefined;
  title: string;
  body: string;
  photoUrl: string;
  wordCount: number;
  articleStatus: string;
  category: string;
  tagsTag: string;
  timePosted: string;
  authorIdentifier: string;
  articleDescpition: string;
}

File: computer_alerts-be/src/main/java/com/calerts/computer_alertsbe/articlesubdomain/presentationlayer/ArticleRequestModel.java
package com.calerts.computer_alertsbe.articlesubdomain.presentationlayer;

import com.calerts.computer_alertsbe.articlesubdomain.dataaccesslayer.ArticleStatus;
import com.calerts.computer_alertsbe.authorsubdomain.datalayer.AuthorIdentifier;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class ArticleRequestModel {

    private String articleId;
    private String title;
    private String body;
    private int wordCount;
    private ArticleStatus articleStatus;
    private String category;
    private String tagsTag;
    private LocalDateTime timePosted;
    private String photoUrl;
    private AuthorIdentifier authorIdentifier;
    private String articleDescription;

}
